### PR TITLE
Added Save Report functionality for Project Performance Dashboard

### PR DESCRIPTION
**Pull request by rickypchen**
@esoergel @czue 

As per spec on [Google Spreadsheet](https://docs.google.com/spreadsheets/d/1hr5-u9Ni90yhumGJqvROeNvqqn0daIPRVLuzCiIbwb8/edit?ts=5760652c#gid=1814269798), users can now save Project Performance Dashboard reports (with filters applied) and create scheduled reports to be sent via email
screen shot 2016-07-27 at 6 00 09 pm

### CODE REVIEW

**Comment by esoergel** [comment]
Merge conflicts.  Let's offline and see if this can use the report framework's "save report" functionality more directly.

**Comment by rickypchen** [comment]
Highlights of changes made:
- Resolved merge conflict
- Async loading for Project Health Dashboard (using report framework)
- Using report framework's "save report" functionality more directly

@esoergel @czue

File: corehq/apps/reports/standard/project_health.py [hunk 1]
```diff
@@ -279,7 +281,6 @@ class ProjectHealthDashboard(ProjectReport):
 
     exportable = True
     emailable = True
-    asynchronous = False
```
Comment (esoergel) [inline corehq/apps/reports/standard/project_health.py]: :tada:

File: corehq/apps/reports/templates/reports/async/project_health_dashboard.html [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+{% block js-inline %} {{ block.super }}
+    <script type="text/template" id="js-template-loading-report">
+        <div class="report-loading-container">
+            <div class="report-loading">
+                <h4>{% trans "Loading Report" %}</h4>
+                <i class="fa fa-spin fa-spinner"></i>
+            </div>
+        </div>
+    </script>
```
Comment (esoergel) [inline corehq/apps/reports/templates/reports/async/project_health_dashboard.html]: Is this script block still necessary?  I'd expect it to be covered by `{{ block.super }}`

**Comment by esoergel** [comment]
These changes all look good, how confident are you that we can deploy this without further testing?  It's still feature flagged, right?

**Comment by rickypchen** [comment]
@esoergel Correct this is still feature flagged, so we can experiment and get some feedback from selected users

**Comment by esoergel** [comment]
Also, for future reference, it's difficult to tell in this view what's changed in the html.  You can make that easier by splitting up the non functional changes from the functional ones.  For instance, one commit which only renames the file, then another commit which moves the `js-inline` block, but still doesn't change anything, and then separately commit the functional changes.  That way people reading the PR can quickly skim over the first two commits and focus more clearly on the meaningful changes.
I know sometimes that's hard to do, just something to keep in mind.

**Comment by esoergel** [comment]
Cool, sounds good to go, then!  Also, this PR adds async functionality and saving, but it's +205 −219.  That's my kinda ratio!
:star:

File: corehq/apps/reports/templates/reports/async/project_health_dashboard.html (hunk 2 above)
Comment (rickypchen) [inline corehq/apps/reports/templates/reports/async/project_health_dashboard.html]: @esoergel Yup, I will remove the script block. It appears to be superfluous

File: corehq/apps/reports/generic.py [hunk 3]
```diff
@@ -437,7 +437,7 @@ def _update_initial_context(self):
 
         def is_datespan(field):
             field_fn = to_function(field) if isinstance(field, basestring) else field
-            return issubclass(field_fn, DatespanFilter)
+            return issubclass(field_fn, DatespanFilter) and field_fn.is_viewable
```
Comment (czue) [inline corehq/apps/reports/generic.py]: is `is_viewable` the right name? the logic between `is_viewable` and `is_datespan` and how they interplay here doesn't quite line up in my head.

**Comment by czue** [comment]
this is great @rickypchen ! really glad to bring this report closer in line with everything else. :clap:

File: corehq/apps/reports/generic.py (hunk 3 above)
Comment (esoergel) [inline corehq/apps/reports/generic.py]: This report does have a datespan, but it shouldn't be editable.  The "save report" modal tries to let you set "last week", "last month", etc, but it should always be "last month".  Is this more comprehensible?  (undecided between "editable" and viewable")

```python
def is_editable_datespan(field):
    field_fn = to_function(field) if isinstance(field, basestring) else field
    return issubclass(field_fn, DatespanFilter) and field_fn.is_editable

has_editable_datespan = any([is_editable_datespan(field) for field in self.fields])
```

The whole business is a little messy - the standard reports strongly suggest that you have a normal, adjustable datespan.  This and the `LastMonthDatespanFilter` are both hacks to get around that.

Comment (czue) [inline corehq/apps/reports/generic.py]: yeah thanks, that makes it much clearer. any reason not to make those name changes if you haven't already?

**Pull request merged**